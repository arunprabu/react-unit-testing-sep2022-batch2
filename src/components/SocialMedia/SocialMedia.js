import React, { useState } from 'react'

const SocialMedia = ({name, followersCount}) => {

  const [ country, setCountry] = useState('Canada');

  const handleCountryChange = (event) => {
    // console.log(event.target.value);
    setCountry(event.target.value);
  }

  return (
    <div>
      <h2>Social Media</h2>
      <h3>Testing Props, Events, States, JSX, CSS, Snapshot Testing</h3>
      <p>Top Social Media: </p>
      <p data-testid="topMediaInfo">{name} with {followersCount} Followers</p>

      <label htmlFor="countryInput">Enter Country Name</label>
      <input type='text' placeholder='Enter Country' value={country} 
      onChange={handleCountryChange} id="countryInput"/>
      <p role='paragraph'>Visit Cognizant {country} website to know more</p>
    </div>
  )
}

export default SocialMedia