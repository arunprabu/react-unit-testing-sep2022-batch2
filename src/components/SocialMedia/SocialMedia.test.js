import { render, screen } from "@testing-library/react";
import SocialMedia from "./SocialMedia";

// group of related test specs -- TEST SUITE
describe('SocialMediaComponent', () => {
  
  // both 'it' and 'test' are same
  it('receives props such as name and followersCount correctly', () => {
    render(<SocialMedia name='Instagram' followersCount='200K'/>);

    const topMediaInfoEl = screen.getByTestId('topMediaInfo');
    expect(topMediaInfoEl.textContent).toBe('Instagram with 200K Followers');
  });

});