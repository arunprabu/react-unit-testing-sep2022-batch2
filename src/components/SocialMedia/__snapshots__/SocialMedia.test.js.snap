// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialMediaComponent has right SocialMedia comp snapshot 1`] = `
<div>
  <h2>
    Social Media
  </h2>
  <h3>
    Testing Props, Events, States, JSX, CSS, Snapshot Testing
  </h3>
  <p>
    Top Social Media: 
  </p>
  <p
    data-testid="topMediaInfo"
    style={
      Object {
        "backgroundColor": "#00ff00",
      }
    }
  >
    Instagram
     with 
    200K
     Followers
  </p>
  <label
    htmlFor="countryInput"
  >
    Enter Country Name
  </label>
  <input
    id="countryInput"
    onChange={[Function]}
    placeholder="Enter Country"
    type="text"
    value="Canada"
  />
  <p
    role="paragraph"
  >
    Visit Cognizant 
    Canada
     website to know more
  </p>
</div>
`;
